lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3OutputStream.java

```diff
@@ -222,6 +230,14 @@ private void flushBuffer(boolean finished)
                 client.putObject(request, RequestBody.fromByteBuffer(bytes));
                 return;
             }
+            catch (S3Exception e) {
+                failed = true;
+                // when `location` already exists, the operation will fail with `412 Precondition Failed`
+                if (e.statusCode() == HTTP_PRECON_FAILED) {
```

Comment:
Is that only case when it fails with this error?
Should you check error message too?

Reply:
yes, at least according to the documentation